src/output_influx.c

```diff
@@ -309,15 +314,20 @@ static void R_API_CALLCONV print_influx_data(data_output_t *output, data_t *data
             data_time = d;
     }
 
-    if (!data_model) {
+    if (influx->metric_format[0])  {
+        char metric_name[1000];
+        mbuf_reserve(buf, sizeof (metric_name) + 1000);
+        expand_topic_string(metric_name, influx->metric_format, data, influx->hostname, influx_sanitize_tag);
+        mbuf_snprintf(buf, "%s", metric_name);
+    } else if (!data_model) {
```

Comment:
I'm not sure about this? `influx->metric_format` is a user setting but `data_model` is an event property, is there a logic to mixing those within a condition?
Also do we guarantee `influx->metric_format` is always present? Otherwise it should be `if (influx->metric_format && *influx->metric_format)`